### PR TITLE
REST v2: return parent commit IDs in `CommitMeta.getParentCommitHashes`

### PR DESCRIPTION
**Pull request by snazy**
Also deprecate `LogEntry.getAdditionalParents()` for removal, it's a duplicate of `CommitMeta.getParentCommitHashes`.

### CODE REVIEW

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java [hunk 1]
```diff
@@ -492,12 +485,15 @@ private ImmutableLogEntry commitToLogEntry(boolean fetchAll, Commit commit) {
...
       Hash hash, CommitMeta commitMeta, List<Hash> additionalParents) {
     ImmutableCommitMeta.Builder updatedCommitMeta = commitMeta.toBuilder().hash(hash.asString());
-    if (additionalParents != null && additionalParents.size() == 1) {
-      // Only add the 1st commit ID. The MERGE_PARENT_PROPERTY was introduced for compatibility
-      // with older clients. There is currently only one use case for the property: exposing the
-      // commit ID of the merged commit.
-      updatedCommitMeta.putProperties(
-          CommitMeta.MERGE_PARENT_PROPERTY, additionalParents.get(0).asString());
+    if (additionalParents != null && !additionalParents.isEmpty()) {
+      additionalParents.forEach(p -> updatedCommitMeta.addParentCommitHashes(p.asString()));
```
Comment (dimas-b) [inline servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java:489]: I think we should add the "proper" parent as the first element of `ParentCommitHashes` and "additional" parents as the following elements. That was the idea behind `CommitMeta.getParentCommitHashes()` (essentially following `git` approach).

Essentially, the "proper" parent of the main lineage is always `getParentCommitHashes().get(0)` and is always present (except for the "root" commit, perhaps). `getParentCommitHashes().get(1)` is the merge parent (in one-way merge cases), `.get(2)` is the second merge parent (no existing use cases for this ATM), etc.

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java [hunk 2]
```diff
@@ -492,12 +485,15 @@ private ImmutableLogEntry commitToLogEntry(boolean fetchAll, Commit commit) {
...
+      if (additionalParents.size() == 1) {
+        // Only add the 1st commit ID. The MERGE_PARENT_PROPERTY was introduced for compatibility
```
Comment (dimas-b) [inline servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java:491]: nit: this comment looks a bit fuzzy... we have only one element in this case to begin with :thinking:

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java (hunk 1 above)
Comment (snazy) [reply servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java:489]: makes sense

File: servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java [hunk 3]
```diff
@@ -489,17 +481,17 @@ private ImmutableLogEntry commitToLogEntry(boolean fetchAll, Commit commit) {
...
-      Hash hash, CommitMeta commitMeta, List<Hash> additionalParents) {
-    ImmutableCommitMeta.Builder updatedCommitMeta = commitMeta.toBuilder().hash(hash.asString());
-    if (additionalParents != null && additionalParents.size() == 1) {
-      // Only add the 1st commit ID. The MERGE_PARENT_PROPERTY was introduced for compatibility
-      // with older clients. There is currently only one use case for the property: exposing the
-      // commit ID of the merged commit.
+  private static CommitMeta enhanceCommitMeta(Hash hash, CommitMeta commitMeta) {
+    if (commitMeta.getParentCommitHashes().size() > 1) {
+      ImmutableCommitMeta.Builder updatedCommitMeta = commitMeta.toBuilder().hash(hash.asString());
+      // Only add the 1st commit ID to the legacy MERGE_PARENT_PROPERTY. It was introduced for
```
Comment (dimas-b) [inline servers/services/src/main/java/org/projectnessie/services/impl/TreeApiImpl.java:487]: nit: `the 1st commit ID` -> `the first merge parent ID`?

File: versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractSingleBranch.java [hunk 4]
```diff
@@ -154,3 +161,12 @@ private List<Operation> singleBranchManyUsersOps(
...
 }
+
+//   [CommitMeta{hash=89ed3813d41d9b3f15ee1b62ea6a41d81743be8a7372e7c047159065c8db4406,
+// committer=null, author=null, allAuthors=[], signedOffBy=null, allSignedOffBy=[], message=user
+// 002/commit 019, commitTime=null, authorTime=null, allProperties={},
+// parentCommitHashes=[35fdf4b429911fd5b24bf45222703600c9fc63f93cae16118cf2d60769428792]},
+//   [CommitMeta{hash=89ed3813d41d9b3f15ee1b62ea6a41d81743be8a7372e7c047159065c8db4406,
+// committer=null, author=null, allAuthors=[], signedOffBy=null, allSignedOffBy=[], message=user
+// 002/commit 019, commitTime=null, authorTime=null, allProperties={},
+// parentCommitHashes=[ec2e96fe02635ef44d101f72df75f1082eb1880cc7eb0f19ef3b0c86709e8460]},
```
Comment (dimas-b) [inline versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractSingleBranch.java:172]: remove this?

**Review by dimas-b — COMMENTED** [review]
LGTM :) just some code clean-up comments

File: versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractSingleBranch.java (hunk 4 above)
Comment (snazy) [reply versioned/tests/src/main/java/org/projectnessie/versioned/tests/AbstractSingleBranch.java:172]: isn't Git a notepad replacement... :trollface:

**Review by dimas-b — APPROVED** [review]

**Pull request merged**